### PR TITLE
Add `Claim all` button with saga

### LINKED ISSUE #3155: Claim All Stakes Button & Saga
### Story

You want to be able to claim all your stakes easily, without having to find and navigate to each individual motion to check if it is ready to be staked or not, then having to perform and wait for each transaction individually as you do so.

### Description

By adding an easily accessible 'Claim all' button, the user will be able to easily claim back their stakes. This feature involves the following:

- Creating a component for the 'Claim all' button section of the 'Stakes' tab.
- Trigger grouped transactions to claim an array of stakes with the `motionId` of each.
- You will also need to update the `claimMotionRewards.ts` saga to be able to handle an array of motion ids.
- Adding the component to the 'Stakes' tab with the "Your stakes" copy

### Design

[Figma link](https://www.figma.com/file/m96yJW1PPXezFl2mkuYha6/Token-Activation?node-id=2411%3A7599)

your-stakes-all-stakes

### This issue is complete when...

- [x] Create a component for the 'Claim all' button section
- [x] Perform the transactions for claiming all staked motions
- [x] Update the `claimMotionRewards.ts` saga
- [x] Add component to 'Stakes' tab with "Your stakes" copy

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
## Description

- Added "Claim all" button to the stakes tab, now you should be able to claim all the stakes there. I tried to simplify the related saga as much as I could, let me know if you come up with something that can be improved.

**New stuff** ✨

* Added `ClaimAllButton` component and connected it to the stakes tab

**Changes** 🏗

* Updated `transform` on `FinalizeMotionAndClaimWidget`
* Updated `claimMotionRewards` to accept multiple motion ids

FireShot Capture 533 - Colony - localhost

Resolves #3155

### CODE REVIEW

**Comment by arrenv** [comment]
Hey @ArmandoGraterol, just confirming, this is only for the 'Claim all' button functionality. Not the copy next to the button or anything else?

**Comment by ArmandoGraterol** [comment]
> Hey @ArmandoGraterol, just confirming, this is only for the 'Claim all' button functionality. Not the copy next to the button or anything else?

Uh I don't think so? Isn't that part of another issue?

**Comment by arrenv** [comment]
@ArmandoGraterol I agree, it is not at all clear in the raised issue. I have updated it, it should include the "Your stakes" copy.

**Comment by ArmandoGraterol** [comment]
> @ArmandoGraterol I agree, it is not at all clear in the raised issue. I have updated it, it should include the "Your stakes" copy.
> 
> I hope that is okay, otherwise, I will create a separate issue for it.

Done! Thank you for the clarification

**Review by arrenv — CHANGES_REQUESTED** [review]
@ArmandoGraterol really nice work. I like how you used the same saga.

I tested and everything seemed to work great. However, I tested it again so that I could do a screen record and the result was the following gif. I will restart my dev environment and test again though.

claimable-transactions-not-working

File: src/modules/users/components/TokenActivation/TokenActivationContent/TokenActivationContent.css [hunk 1]
```diff
@@ -282,3 +282,35 @@
...
+  width: 100%;
+}
+
+.claimAllButtonSection button {
+  width: 100px;
+  font-size: --size-small;
+}
+
+.claimAllButtonSection h4 {
+  font-size: --size-smallish;
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/modules/users/components/TokenActivation/TokenActivationContent/TokenActivationContent.css:311]: Oh damn you are right! Thank you

**Comment by ArmandoGraterol** [comment]
> I tested and everything seemed to work great. However, I tested it again so that I could do a screen record and the result was the following gif. I will restart my dev environment and test again though.

Please do so! I haven't been able to reproduce that

**Comment by arrenv** [comment]
@ArmandoGraterol I have been testing this some more and the issue is with the `useClaimableStakedMotionsQuery` query.
Curerntly, you have to reload the app for the query to update, and it also includes motions that have been staked and not finalized.

I will create a separate issue for updating the resolver.

With this PR, do you think there needs to be feedback or fallback for if the transaction fails or can't be performed. For example, in the Network tab in console, when my gif above happens, I get this error.

`message: "VM Exception while processing transaction: revert voting-rep-motion-not-claimable"`

**Comment by arrenv** [comment]
@ArmandoGraterol I have updated the resolver, which should help with this issue - https://github.com/JoinColony/colonyDapp/pull/3175

**Comment by ArmandoGraterol** [comment]
> @ArmandoGraterol I have updated the resolver, which should help with this issue - #3175

Nice! I also added an extra, just in case, error that was missing from the saga, and added a query to update the claim list after claiming something with the saga. So, the case in your gif should now not happen

**Review by arrenv — COMMENTED** [review]
@ArmandoGraterol It is working great, nice!

With the changes I made to `useClaimableStakedMotionsQuery`  in https://github.com/JoinColony/colonyDapp/pull/3175 returning motions as opposed to just an array of motionIds, should we manage that in this PR or do another PR for it?

**Comment by ArmandoGraterol** [comment]
> With the changes I made to `useClaimableStakedMotionsQuery` in #3175 returning motions as opposed to just an array of motionIds, should we manage that in this PR or do another PR for it?

I think it's more convenient if you merge yours first and then I update this one with the changes you made on that PR :)

**Review by chinins — APPROVED** [review]

**Review by chinins — APPROVED** [review]
Very nice work. 
I noticed that after claiming all I still have value in the staked pat of the token activation popover which resets to zero after I refresh. I think it would be good to refresh the query that provides these numbers (I think it's user lock query)  in the claimMotionRewards saga after the action success.
Screenshot 2022-02-14 at 23 55 38

**Comment by ArmandoGraterol** [comment]
> I noticed that after claiming all I still have value in the staked pat of the token activation popover which resets to zero after I refresh. I think it would be good to refresh the query that provides these numbers (I think it's user lock query) in the claimMotionRewards saga after the action success.

Oh yeah, I agree with you! I Thank you for bringing that up

File: src/modules/users/components/TokenActivation/TokenActivationContent/StakesTab.tsx [hunk 2]
```diff
@@ -4,10 +4,17 @@ import { FormattedMessage, defineMessages } from 'react-intl';
...
 import { useClaimableStakedMotionsQuery } from '~data/generated';
+import Heading from '~core/Heading';
+
+import ClaimAllButton from './ClaimAllButton';
 
 import styles from './TokenActivationContent.css';
 
 const MSG = defineMessages({
+  tabTitle: {
+    id: 'users.TokenActivation.TokenActivationContent.ClaimsTab.noClaims',
```
Comment (danbr) [inline src/modules/users/components/TokenActivation/TokenActivationContent/StakesTab.tsx:15]: .ClaimsTab..... - but was renamed to StakesTab. I can update when i make my changes for the dialog.

**Review by danbr — APPROVED** [review]
nice work - was good for me to review, as i'll be rebasing off your PR and incorporating your changes with my PR.

**Pull request merged**